src/Dfc.ProviderPortal.Apprenticeships/Helper/CosmosDbHelper.cs

```diff
@@ -386,5 +391,68 @@ public async Task<int> GetTotalLiveApprenticeships(DocumentClient client, string
                 .Where(cr => cr.RecordStatus == RecordStatus.Live)
                 .CountAsync();
         }
+             
+      
+       public async Task<int> ExecuteStoredProcedureAsync(DocumentClient client, string collectionId, string procedureName, int UKPRN, int currentStatus, int statusToBeChangedTo, int partitionKey)
+        {          
+            RequestOptions requestOptions = new RequestOptions { PartitionKey = new PartitionKey(partitionKey), EnableScriptLogging = true };         
+
+            var response =  await client.ExecuteStoredProcedureAsync<int>(UriFactory.CreateStoredProcedureUri(_settings.DatabaseId, collectionId, "Apprenticeship_ChangeRecordStatus"), requestOptions, UKPRN, currentStatus, statusToBeChangedTo);
+                 
+            return response;
+
+        }
+        public async Task Initialise_CreateStoreProc()
+        {       
+            string scriptFileName = @"Data/StoreProcedureApprenticeship_ChangeRecordStatus.js";
+            string StoredProcedureName = Path.GetFileNameWithoutExtension(scriptFileName);          
+
+            await CreateStoredProcedureIfNotExistsAsync(GetClient(), _settings.DatabaseId, StoredProcedureName, scriptFileName);
+        }
+      
+        public async Task CreateStoredProcedureIfNotExistsAsync(DocumentClient client, string collectionId, string procedureName, string procedurePath)
+        {
+            var errorMessage = String.Empty;
+            string sProcresult;
+            Throw.IfNull(client, nameof(client));
+            Throw.IfNullOrWhiteSpace(collectionId, nameof(collectionId));
+
+            string scriptFileName = @"/Data/StoreProcedure/Apprenticeship_ChangeRecordStatus";
+            string StoredProcedureName = Path.GetFileNameWithoutExtension(scriptFileName);
+          
+            var collectionLink = string.Join(@",", UriFactory.CreateDocumentCollectionUri(_settings.DatabaseId, "apprenticeship")  + "/sprocs/");
+
+            StoredProcedure isStoredProcedureExist = client.CreateStoredProcedureQuery(collectionLink)
+                                   .Where(sp => sp.Id == StoredProcedureName)
+                                   .AsEnumerable()
+                                   .FirstOrDefault();                        
+            try
+            {
+                if (isStoredProcedureExist == null)
+                {
+                    Assembly assembly = this.GetType().Assembly;
+                    var resourceStream = assembly.GetManifestResourceStream(assembly.GetName().Name + "." + "Data.StoreProcedure" + ".Apprenticeship_ChangeRecordStatus.js");
```

Comment:
Need a `using` statement around this stream

Reply:
Done

Reply:
Done